libraries/Microsoft.Bot.Configuration/BotConfigurationExtensions.cs

```diff
@@ -0,0 +1,153 @@
+            if (botServices.Count == 0)
+            {
+                return Enumerable.Empty<ConnectedService>();
+            }
+
+            // Obtains all the services of the specified type
+            var botFilteredServices = botServices.Where(service => service.Type.Equals(serviceType, StringComparison.CurrentCultureIgnoreCase));
+
+            // If there isn't any service of the specified type, returns an empty IEnumerable
+            if (botFilteredServices == null || botFilteredServices.Count() == 0)
+            {
+                return Enumerable.Empty<ConnectedService>();
+            }
+
+            return botFilteredServices;
+        }
+
+        /// <summary>
+        /// Check if the Project's Name contains white spaces.
+        /// </summary>
+        /// <param name="botConfiguration">BotConfiguration object to extend.</param>
+        /// <param name="folder">Folder's path to validate.</param>
+        /// <exception cref="System.Exception"> Thrown when no '.scproj' file is found or when there are two or more 'csproj' files.</exception>
+        /// <exception cref="System.ArgumentNullException"> Thrown when param 'folder' is null.</exception>
+        /// <returns>Boolean that indicates if a Project's Name is spaceless.</returns>
+        public static bool IsProjectNameSpaceless(this BotConfiguration botConfiguration, string folder)
+        {
+            // If the parameter 'folder' is null or empty, throws an Exception
+            if (string.IsNullOrEmpty(folder))
+            {
+                throw new ArgumentNullException(nameof(folder));
+            }
+
+            var files = Directory.GetFiles(folder, "*.csproj", SearchOption.TopDirectoryOnly);
+
+            // Throws an error if there are two or more 'csproj' files
+            if (files.Length > 1)
+            {
+                throw new Exception("There could not be more than one '.csproj' file.");
+            }
+
+            try
+            {
+                // Gets the path to the first (and only) file with extension 'csproj'.
+                var file = files.FirstOrDefault();
+
+                // Checks if the file name contains any white space
+                return !file.Contains(" ");
+            }
+            catch (FileNotFoundException ex)
+            {
+                // If there's no file, throws an Exception
+                throw new FileNotFoundException("There isn't any \'.csproj\' in the specified folder", ex);
+            }
+        }
+
+        /// <summary>
+        /// Checks if the <see cref="LuisService"/> has its key defined.
+        /// </summary>
+        /// <param name="botConfiguration">BotConfiguration object to extend.</param>
+        /// <returns>Tuple with a boolean that indicates if a bot configuration file's Luis keys are present and its Error message.</returns>
+        public static Tuple<bool, string> ValidateLuisKeyExistence(this BotConfiguration botConfiguration)
```

Comment:
Suggest using newer C#7 tuple syntax. 

NOTE: applies to all `Tuple<>` usage.

Reply:
+1 on the C# 7 tuple syntax. We're already using that in a number of other places.